### ISSUE TITLE
Switch expressions do not correctly infer nullability

### ISSUE THREAD

**Issue by Arshia001** [dotnet/roslyn#34828]
**Version Used**: 
Visual Studio 2019 version 16.0.0 + dotnet core version 3.0.100-preview3-010431

**Steps to Reproduce**:

```
void F(string s) { }
string? str = ...;
F(str switch { null => "NULL", _ => str });
```

**Expected Behavior**: The call to `F` succeeds, since we already know that str is non-null in the second arm of the switch expression.

**Actual Behavior**: The call to F fails since the result of the switch expression is determined to be of type `string?`.

**Comment by gafter** [comment]
This is likely a dup of #34233 which will be fixed by #34686.  Once that is integrated I will double-check.

**Comment by gafter** [comment]
Verified fixed in master.  Fix should appear in VS 2019 Update 1 Preview 3.
